src/main/scala/com/microsoft/hyperspace/index/rules/JoinIndexRule.scala

```diff
@@ -518,8 +514,12 @@ object JoinIndexRule extends Rule[LogicalPlan] with Logging {
       allRequiredCols: Seq[String]): Seq[IndexLogEntry] = {
     indexes.filter { idx =>
       val allCols = idx.indexedColumns ++ idx.includedColumns
-      requiredIndexCols.toSet.equals(idx.indexedColumns.toSet) &&
-      allRequiredCols.forall(allCols.contains)
+
+      // All required index columns should match one-to-one with all indexed columns and
+      // vice-versa. All required columns must be present in the available index columns.
+      isResolved(spark, requiredIndexCols, idx.indexedColumns) &&
+      requiredIndexCols.distinct.size == idx.indexedColumns.distinct.size &&
```

Comment:
distinct needs to take care of case sensitivity or no?

Reply:
it should. Thanks for the catch. Let me fix that

Reply:
meaning we don't have a test coverage for this?

Reply:
yeah, I will add a test where join condition is
```sql
T1.a = T2.a and T1.A = T2.B
```
where `T1.a` is the same as `T1.A`
and it should not pick index. WIth the current (bug) logic, this case will pick any T1 index with 2 columns, one being `a`

Reply:
cool thanks!